docs/introduction/container-images.md

```diff
@@ -0,0 +1,250 @@
+
+ - *cgroups*: controls resources within the kernel (io, cpu, devices, memory, network). There is something called a "freezer" that you put a bunch of processes into and put them to sleep without worrying about resources between them.
+ - *namespaces*: provide a pure isolation layer inside the kernel. Most of the time, resources only belong to a single one, for example:
+   - networking devices go in networking namespaces
+   - ipc namespace for messages between containers 
+   - mount namespace for filesystem tree
+   - pid namespace - init in linux needs to be running as pid 1 it isolates the process tree between containers     
+   - UTS namespace because each container needs a separate hostname, 
+   - User namespace is important for "pretending to be root" without being root.)
+
+For those interested, there is a good talk about the history of the API development [here](https://www.youtube.com/watch?v=YsYzMPptB-k#t=18m10s).
+In early containers, you could break from a container and become root on the host. This is bad, very bad.
+These core components are the same for all systems! They are unfortuntately very complex.
+
+A fun history fact (and maybe you already figured this about because you watched the video links above!). These two things (cgroups and namespaces) were developed by different open source groups. This came from an agreement at a conference in 2011 called the [Kernel Summit](https://lwn.net/Articles/KernelSummit2011/), literally a bunch of people sitting in a room that had both developed technologies they wanted added to the linux kernel, and had to go through component by component and argue their case for their thing. At the end we did get a single cohesive thing, but it had these very two different cgroups and namespaces. 
+
+For the applied amount us, the definitions of "namespaces" and "cgroups" is not satisfactory.
+I don't get it, show me! For a typical Linux OS (the example shown below is Ubuntu 16.04), 
+these ideas map to folders on your computer, and 
+the APIs generally work to create and modify content here, and dictate rules (permissions)
+for interactions.
+
+**Where are the namespaces?**
+
+Here are the namespaces:
+
+```
+$ ls -l /proc/self/ns/
+total 0
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 cgroup -> cgroup:[4026531835]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 ipc -> ipc:[4026531839]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 mnt -> mnt:[4026531840]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 net -> net:[4026532009]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 pid -> pid:[4026531836]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 pid_for_children -> pid:[4026531836]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 user -> user:[4026531837]
+lrwxrwxrwx 1 vanessa vanessa 0 Sep 11 12:42 uts -> uts:[4026531838]
+```
+
+**Where are the cgroups?**
+
+Here! Each cgroup is separately mountable. 
+
+```bash
+ls -l /sys/fs/cgroup
```

Comment:
this won't age well :-)
Cgroup v2 is coming imminently and has a different layout.

Reply:
Let' cut it out then.